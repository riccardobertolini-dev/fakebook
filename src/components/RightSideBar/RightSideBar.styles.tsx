import styled from 'styled-components';

export const RightSideMenu = styled.div`
	max-width: 200px;
	width: 100%;
	margin: 40px auto 0 0;
	display: none;

	@media screen and (min-width: 1024px) {
		max-width: 200px;
		display: block;
	}
	@media screen and (min-width: 1220px) {
		max-width: 300px;
	}
`;

export const SideSection = styled.div`
	background-color: #fff;
	border: 1px solid #ccc;
	font-family: 'Lato', sans-serif;
	font-weight: 400;
	font-size: 13px;
	line-height: 1.6;
	padding: 10px;
	border-radius: 10px;
	margin-bottom: 20px;

	h3 {
		border-bottom: 1px solid #ccc;
		font-size: 18px;
	}

	p {
		font-size: 14px;
	}

	span {
		margin-left: 10px;
		font-size: 10px;
	}
`;

export const Accordion = styled.div`
	cursor: pointer;
`;

export const AccordionText = styled.div<{ accordionOpen: boolean }>`
	display: ${props => (props.accordionOpen ? 'block' : 'none')};
`;
