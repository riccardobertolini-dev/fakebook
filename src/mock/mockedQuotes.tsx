export const MockedQuote1 = {
	text: 'Genius is one percent inspiration and ninety-nine percent perspiration.',
	author: 'Thomas Edison',
};
export const MockedQuote2 = {
	text: 'You can observe a lot just by watching.',
	author: 'Yogi Berra',
};
