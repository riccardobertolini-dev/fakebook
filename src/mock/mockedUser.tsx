export const MockedUser1 = {
	gender: 'female',
	name: {
		title: 'Ms',
		first: 'Isabelle',
		last: 'Davies',
	},
	location: {
		street: {
			number: 8255,
			name: 'Esk Street',
		},
		city: 'Taupo',
		state: 'West Coast',
		country: 'New Zealand',
		postcode: 25207,
		coordinates: {
			latitude: '53.3834',
			longitude: '16.3516',
		},
		timezone: {
			offset: '-6:00',
			description: 'Central Time (US & Canada), Mexico City',
		},
	},
	email: 'isabelle.davies@example.com',
	login: {
		uuid: '71d3d4bf-2c47-450f-b3bf-e15a7fef23da',
		username: 'yellowtiger203',
		password: 'pyramid',
		salt: 'WrEua7np',
		md5: '8d185b252bdc6739cc36a7d4c67bcb1f',
		sha1: '4843e0607dd75b952504e278013508847f002237',
		sha256: 'dde6119b6dad7740579c5ace7f89789b7c24973dd203285bac2028018a046926',
	},
	dob: {
		date: '1965-10-08T18:06:10.878Z',
		age: 57,
	},
	registered: {
		date: '2015-12-03T15:58:10.458Z',
		age: 6,
	},
	phone: '(999)-128-6616',
	cell: '(132)-372-7000',
	id: {
		name: '',
		value: null,
	},
	picture: {
		large: 'https://randomuser.me/api/portraits/women/44.jpg',
		medium: 'https://randomuser.me/api/portraits/med/women/44.jpg',
		thumbnail: 'https://randomuser.me/api/portraits/thumb/women/44.jpg',
	},
	nat: 'NZ',
};
