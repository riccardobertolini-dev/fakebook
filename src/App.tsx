import React, { useState } from 'react';
import { Routes, Route } from 'react-router-dom';
import HomePage from './layouts/HomePage/HomePage';
import Profile from './layouts/Profile/Profile';
import { GlobalContext } from './globalContext';
import { User } from './models/User';
import { SideChatUsers, Logo, NavBar, Root, SideChat, UserStatus, UserThumbnail, InnerThumbnail, BodyComponent, CentralContent } from './App.styles';
import { SocialWallTile } from './components/UserTile/UserTile.styles';
import { RightSideBar } from './components/RightSideBar/RightSideBar';

function App() {
	const [users, updateUsers] = useState<User[]>([]);
	const [userPosts, updateUserPosts] = useState<string[]>([]);

	return (
		<Root>
			<GlobalContext.Provider value={{ users, updateUsers, userPosts, updateUserPosts }}>
				<NavBar>
					<Logo to={'/'}>
						<span>FAKE</span>BOOK
					</Logo>
				</NavBar>
				<BodyComponent>
					<SideChat>
						<SocialWallTile>
							<SideChatUsers>
								<h3>Contacts</h3>
								{users &&
									users.map((user, index) => (
										<UserThumbnail key={`${user.id.value}${index}`}>
											<InnerThumbnail to={`/${user.login.uuid}`}>
												<UserStatus active={Math.random() < 0.5} />
												<img src={user.picture.thumbnail} alt="" />
												{user.name.first} {user.name.last}
											</InnerThumbnail>
										</UserThumbnail>
									))}
							</SideChatUsers>
						</SocialWallTile>
					</SideChat>
					<CentralContent>
						<Routes>
							<Route path="/" element={<HomePage />} />
							<Route path="/:userSlug" element={<Profile />} />
						</Routes>
					</CentralContent>
					<RightSideBar />
				</BodyComponent>
			</GlobalContext.Provider>
		</Root>
	);
}

export default App;
