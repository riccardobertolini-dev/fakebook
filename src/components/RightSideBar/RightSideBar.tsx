import { RightSideMenu, SideSection } from './RightSideBar.styles';
import { AiOutlineGift } from 'react-icons/ai';
import React from 'react';
import { useGlobalContext } from '../../globalContext';
import { User } from '../../models/User';

export const generateRandomBirthday = (users: User[], index: number) => {
	return `${Math.random() > 0.5 ? 'Tomorrow' : 'Today'} is ${users[index].name.first}'s Birthday`;
};

export const RightSideBar = () => {
	const { users } = useGlobalContext();

	return users.length > 0 ? (
		<RightSideMenu>
			<SideSection>
				<h3>Birthdays</h3>
				<div>
					<AiOutlineGift /> {generateRandomBirthday(users, 1)}
				</div>
				<div>
					<AiOutlineGift /> {generateRandomBirthday(users, 2)}
				</div>
				<div>
					<AiOutlineGift /> {generateRandomBirthday(users, 3)}
				</div>
			</SideSection>
		</RightSideMenu>
	) : null;
};
