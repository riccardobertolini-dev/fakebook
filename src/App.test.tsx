import * as React from 'react';
import renderer from 'react-test-renderer';
import App from './App';
import { Routes, Route } from 'react-router-dom';
import { MockedUser1 } from './mock/mockedUser';
import { generateRandomBirthday, RightSideBar } from './components/RightSideBar/RightSideBar';

jest.mock('react-router-dom', () => ({
	Routes: jest.fn(),
	Route: jest.fn(),
	Link: jest.fn(),
}));

jest.mock('./components/RightSideBar/RightSideBar', () => ({
	RightSideBar: () => <div></div>,
	generateRandomBirthday: () => 'Hello',
}));

jest.mock('axios');

describe('App component', () => {
	afterAll(() => {
		jest.clearAllMocks();
	});

	it('should renders', () => {
		const setStateMock = jest.fn();
		<RightSideBar />;
		jest.spyOn(React, 'useState').mockImplementation(() => [[MockedUser1, MockedUser1, MockedUser1], setStateMock]);
		const wrapper = renderer.create(<App />).toJSON();

		expect(wrapper).toMatchSnapshot();
	});
});
