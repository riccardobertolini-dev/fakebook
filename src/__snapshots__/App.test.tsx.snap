// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should renders 1`] = `
<div
  className="sc-bcXHqe hICBOO"
>
  <div
    className="sc-hLBbgP cggtmG"
  />
  <div
    className="sc-ipEyDJ gHALup"
  >
    <div
      className="sc-jSUZER kjWGah"
    >
      <div
        className="sc-eDWCr ckUNIy"
      >
        <ul
          className="sc-iBYQkv igfonv"
        >
          <li
            className="sc-ftTHYK hJbqbM"
          />
          <li
            className="sc-ftTHYK hJbqbM"
          />
          <li
            className="sc-ftTHYK hJbqbM"
          />
        </ul>
      </div>
    </div>
    <div
      className="sc-csuSiG hveveN"
    />
    <div />
  </div>
</div>
`;
